maven/README.md

```diff
@@ -5,139 +5,220 @@ This Task can be used to run a Maven build.
 ## Install the Task
 
 ```
-kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/maven/maven.yaml
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/maven/maven.yaml
 ```
 
 ## Parameters
 
 - **GOALS**: Maven `goals` to be executed
-- **MAVEN_MIRROR_URL**: Maven mirror url (to be inserted into ~/.m2/settings.xml)
+- **MAVEN_MIRROR_URL**: Maven mirror url (to be inserted into ~/.m2/settings.xml)   
+- **PROXY_USER**: Username to login to the proxy server (to be inserted into ~/.m2/settings.xml)
+- **PROXY_PASSWORD**: Password to login to the proxy server (to be inserted into ~/.m2/settings.xml)
+- **PROXY_HOST**: Hostname of the proxy server (to be inserted into ~/.m2/settings.xml)
+- **PROXY_NON_PROXY_HOSTS**: Non proxy hosts to be reached directly bypassing the proxy (to be inserted into ~/.m2/settings.xml)
+- **PROXY_PORT**: Port number on which the proxy port listens (to be inserted into ~/.m2/settings.xml)
+- **PROXY_PROTOCOL**: http or https protocol whichever is applicable (to be inserted into ~/.m2/settings.xml)
 
 ## Workspaces
 
-* **source**: `git`-type `PipelineResource` specifying the location of the source to build.
+* **source**: `PersistentVolumeClaim`-type so that volume can be shared among `git-clone` and `maven` task
+
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: maven-source-pvc
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 500Mi
+```
 
 ## Usage
 
-This TaskRun runs a Maven build
+This Pipeline and PipelineRun runs a Maven build
 
 ### With Defaults
 
 ```
 apiVersion: tekton.dev/v1beta1
-kind: TaskRun
+kind: Pipeline
+metadata:
+  name: maven-test-pipeline
+spec:
+  workspaces:
+    - name: shared-workspace
+    - name: maven-settings
+  tasks:
+    - name: fetch-repository
+      taskRef:
+        name: git-clone
+      workspaces:
+        - name: output
+          workspace: shared-workspace
+      params:
+        - name: url
+          value: https://github.com/spring-projects/spring-petclinic
+        - name: subdirectory
+          value: ""
+        - name: deleteExisting
+          value: "true"
+    - name: maven-run
+      taskRef:
+        name: maven
+      runAfter:
+        - fetch-repository
+      workspaces:
+        - name: maven-settings
+          workspace: maven-settings
+        - name: source
+          workspace: shared-workspace
+```
+```
```

Comment:
could use `---` and keep it in the same code block :stuck_out_tongue:

Reply:
Okay...Will keep in mind. Actually I created two separate code blocks so that one can easily differentiate :sweat_smile: in the docs